crates/manifest/src/schema/v2.rs

```diff
@@ -265,6 +309,92 @@ mod tests {
             .unwrap();
     }
 
+    #[test]
+    fn deserializing_labels() {
+        AppManifest::deserialize(toml! {
+            spin_manifest_version = 2
+            [application]
+            name = "trigger-configs"
+            [[trigger.fake]]
+            something = "something else"
+            [component.fake]
+            source = "dummy"
+            key_value_stores = ["default", "snake_case", "kebab-case"]
+            sqlite_databases = ["default", "snake_case", "kebab-case"]
+        })
+        .unwrap();
+    }
+
+    #[test]
+    fn deserializing_labels_fails_for_non_kebab_or_snake() {
+        assert!(AppManifest::deserialize(toml! {
+            spin_manifest_version = 2
+            [application]
+            name = "trigger-configs"
+            [[trigger.fake]]
+            something = "something else"
+            [component.fake]
+            source = "dummy"
+            key_value_stores = ["b@dlabel"]
+        })
+        .is_err());
+    }
+
+    fn get_test_manifest_with_labels(labels: Vec<String>) -> AppManifest {
```

Comment:
`Component` is `Serialize` - might be more concise to test with that rather than creating a whole manifest?